tests/handlers/external_test.py

```diff
@@ -189,12 +191,37 @@ async def test_timeseries_detail(client: AsyncClient) -> None:
 
 
 @pytest.mark.asyncio
-async def test_links(
+async def test_links_gcs(
     client: AsyncClient, mock_butler: MockButler, mock_google_storage: None
+) -> None:
+    label = "label-gcs"
+    url = f"https://example.com/{str(mock_butler.uuid)}"
+
+    await _test_links(client, mock_butler, label, url)
+
+
+@pytest.mark.asyncio
+async def test_links_s3(
+    client: AsyncClient, mock_butler: MockButler, s3: boto3.client
+) -> None:
+    label = "label-s3"
+
+    expires = timedelta(hours=1)
+    url = s3.generate_presigned_url(
+        "get_object",
+        Params={"Bucket": "some-bucket", "Key": str(mock_butler.uuid)},
+        ExpiresIn=expires.total_seconds(),
+    )
+
+    await _test_links(client, mock_butler, label, url)
+
+
+async def _test_links(
+    client: AsyncClient, mock_butler: MockButler, label: str, url: str
 ) -> None:
     r = await client.get(
         "/api/datalink/links",
-        params={"iD": f"butler://label/{str(mock_butler.uuid)}"},
+        params={"id": f"butler://{label}/{str(mock_butler.uuid)}"},
```

Comment:
The capitalization here was intentional to test case-insensitivity for that parameter because it's required for IVOA standards compliance (I really should have added a comment or something to say that).

Reply:
Okay I figured that might be the case, but yeah I was confused without a comment